connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java

```diff
@@ -489,6 +490,9 @@ private WorkerTask buildWorkerTask(ClusterConfigState configState,
             OffsetStorageWriter offsetWriter = new OffsetStorageWriter(offsetBackingStore, id.connector(),
                     internalKeyConverter, internalValueConverter);
             Map<String, Object> producerProps = producerConfigs(config);
+            String clientId = Objects.toString(producerProps.get(ProducerConfig.CLIENT_ID_CONFIG), "");
```

Comment:
I wonder if it would be better to add consumer.client.id and producer.client.id to WorkerConfig, and then make this:

```suggestion
            String clientId = config.getString(WorkerConfig.CONSUMER_CLIENT_ID);
```
?

Reply:
Possibly, though the same argument could be made for the other client configuration properties so might get messy.  It actually looks like all the other defaults for client properties are being set in `producerConfigs(...)` and `consumerConfigs(...)`, so it might be tidier to just follow the pattern set for `ConsumerConfig.GROUP_ID_CONFIG` here:  https://github.com/apache/kafka/blob/aff02944fee9dd79e58a6dd31291d5058bc786bb/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java#L539

I would also need to update `producerConfigs(...)` to take a `ConnectorTaskId` as the first argument, following the pattern for `consumerConfigs(...)`.

Reply:
I've made the above change, hopefully the result is more consistent with existing code.